extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java

```diff
@@ -399,14 +403,26 @@ private Page getNextPageFrom(final JsonArray contents)
     private void collectStreamsFrom(@Nonnull final StreamInfoItemsCollector collector,
                                     @Nonnull final JsonArray videos) {
         final TimeAgoParser timeAgoParser = getTimeAgoParser();
-
         videos.stream()
                 .filter(JsonObject.class::isInstance)
                 .map(JsonObject.class::cast)
-                .filter(video -> video.has(PLAYLIST_VIDEO_RENDERER))
-                .map(video -> new YoutubeStreamInfoItemExtractor(
-                        video.getObject(PLAYLIST_VIDEO_RENDERER), timeAgoParser))
-                .forEachOrdered(collector::commit);
+                .forEach(video -> {
+                    if (video.has(PLAYLIST_VIDEO_RENDERER)) {
+                        collector.commit(new YoutubeStreamInfoItemExtractor(
+                                video.getObject(PLAYLIST_VIDEO_RENDERER), timeAgoParser));
+                    } else if (video.has(RICH_ITEM_RENDERER)) {
+                        final JsonObject richItemRenderer = video.getObject(RICH_ITEM_RENDERER);
+                        if (richItemRenderer.has("content")) {
+                            final JsonObject richItemRendererContent =
+                                    richItemRenderer.getObject("content");
+                            if (richItemRendererContent.has(REEL_ITEM_RENDERER)) {
+                                collector.commit(new YoutubeReelInfoItemExtractor(
+                                        richItemRendererContent.getObject(REEL_ITEM_RENDERER),
+                                        timeAgoParser));
+                            }
+                        }
+                    }
```

Comment:
Shouldn't an error be added to the collector to signal that an item in the list was in an unrecognized format? Or is there other garbage inside the list that just needs to be ignored?

Reply:
I am not sure, as there should be playlists video items only but YouTube return(ed?) suggestions of videos to add to users playlists when their owner(s) browse the playlists and is(are) logged with their account(s), so I don't know what elements they use in this case.

Reply:
Ok let's leave it this way and if users report of missing item take a look again